config/src/main/java/com/typesafe/config/impl/ConfigBeanImpl.java

```diff
@@ -163,11 +164,14 @@ private static Object getValue(Class<?> beanClass, Type parameterType, Class<?>
             return config.getList(configPropName);
         } else if (hasAtLeastOneBeanProperty(parameterClass)) {
             return createInternal(config.getConfig(configPropName), parameterClass);
+        } else if (parameterClass.isEnum()) {
+            return Enum.valueOf((Class<Enum>)parameterClass, config.getString(configPropName));
```

Comment:
This should probably handle IllegalArgumentException and convert it to a ConfigException comparable to what happens if you getInt on a string and the string isn't a valid int.

In fact - what do you think of adding Config.getEnum which would take the enum class and return a value; then here you'd just call getEnum rather than getString ?